apps/node-status/src/fetching/refetch-hook.ts

```diff
@@ -0,0 +1,34 @@
+import { useRevalidator } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+
+const REFETCH_INTERVAL = 5000;
+
+export const useRefetchStatusOnInterval = () => {
+  const { revalidate } = useRevalidator();
+
+  useEffect(() => {
+    const interval = setInterval(revalidate, REFETCH_INTERVAL);
+    return () => clearInterval(interval); // Clear the interval when the component is unmounted
+  }, [revalidate]);
+};
+
+const VISIBILITY_STATE_CHANGE_DELAY = 800;
+
+// Meant to slow down the state transition from loading to idle so the UI can show a discernible spinner
+export const useDeceleratedFetchState = () => {
```

Comment:
This hook name threw me off a bit. What about something like `useDelayedIsLoading`? Hm... maybe it's not better. It just took me a minute to realize what this hook was actually doing.

Reply:
Renamed 👍